### ISSUE TITLE
Segmentation fault (core dumped) passing `core._Field` as default value to `defstruct`

### ISSUE THREAD

**Issue by peterschutt** [jcrist/msgspec#342]
### Description

Hi, have been enjoying using msgspec so far, thanks for creating and maintaining it!

I've encountered this:

```py
from msgspec import defstruct, field

print(defstruct("A", [("a", int)]))
print(defstruct("B", [("b", int, 1)]))
print(defstruct("C", [("c", int, field(default=1))]))
print(defstruct("D", [("d", int, field())]))
```
output:

```
<class '__main__.A'>
<class '__main__.B'>
<class '__main__.C'>
Segmentation fault (core dumped)
```

environment:

```
(starlite-py3.8) peter@peter-Inspiron-15-5510:~/PycharmProjects/starlite$ python -c "import msgspec;print(msgspec.__version__)"
0.13.1
(starlite-py3.8) peter@peter-Inspiron-15-5510:~/PycharmProjects/starlite$ python --version
Python 3.8.15
```

background:

I'm generating `_Field` instances from this function:

```py
def _create_msgspec_field(field_definition: FieldDefinition) -> MsgspecField:
    kws: dict[str, Any] = {}
    if field_definition.default is not Empty:
        kws["default"] = field_definition.default

    if field_definition.default_factory is not Empty:
        kws["default_factory"] = field_definition.default_factory

    return field(**kws)
```
So if an upstream `FieldDefinition` doesn't define `default` or `default_factory` the `field()` call is made with no kwargs.

**Comment by jcrist** [comment]
Thanks for the excellent issue report, this has been fixed in #343.

For now you can either [use a dev build](https://jcristharif.com/msgspec/install.html#installing-from-github), or workaround this by avoiding using a field with no args (`field()`). I hope to get a release out soon (hopefully within a week :crossed_fingers: ), but no promises.

**Comment by peterschutt** [comment]
No worries, thanks for the quick fix.

> I hope to get a release out soon (hopefully within a week crossed_fingers ), but no promises.

I have worked around it, so no pressure from here:)

Cheers!
